### PR TITLE
feat: add BKTClient

### PR DESCRIPTION
**Pull request by yshrsmz**
I'll add a periodic-request feature(fetch_evaluations and register_events) later

### CODE REVIEW

File: test/BKTClient.spec.ts [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+  suite('jsonVariation', () => {
+    const JSON_VALUE = '{"key": "value"}'
+
+    test.each([
+      [JSON_VALUE, {}, JSON.parse(JSON_VALUE)],
+      ['true', JSON.parse(JSON_VALUE), true],
+      ['true', {}, true],
+      ['not bool', {}, {}],
+      ['1', {}, 1],
+      ['{}', {}, {}],
```
Comment (yshrsmz) [inline test/BKTClient.spec.ts:383]: do we need `jsonVariation`/`numberVariation`/`booleanVariation` in js SDK?

As you can see from the test case, `JSON.parse()` can parse boolean string and number string. so the behavior here is not consistent with Android SDK.

(also, previous bucketeer-web-sdk only has `stringVariation()` function)

File: src/BKTClient.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      )
+  }
+
+  currentUser(): BKTUser {
+    return toBKTUser(this.component.userHolder().get())
+  }
+
+  updateUserAttributes(attributes: Record<string, string>): void {
+    this.component.userHolder().updateAttributes((_prev) => ({ ...attributes }))
+    this.component.evaluationInteractor().clearCurrentEvaluationsId()
```
Comment (yshrsmz) [inline src/BKTClient.ts:104]: reset currentEvaluationsId when `updateUserAtributes()` is called.

File: test/BKTClient.spec.ts (hunk 1 above)
Comment (cre8ivejp) [reply test/BKTClient.spec.ts:383]: I want to unify the interfaces as much as I can. So, for now, let's keep it as it is.

**Review by cre8ivejp — APPROVED** [review]
Thank you!!

**Pull request merged**